traitsui/qt4/ui_modal.py

```diff
@@ -155,7 +155,7 @@ def init(self, ui, parent, style):
                         self.control.accept,
                         default=default,
                     )
-                    ui.on_trait_change(self._on_error, "errors", dispatch="ui")
+                    ui.observe(self._on_error, "errors", dispatch="ui")
```

Comment:
Looks like we have two listeners that don't get unhooked in this module. Instead of going over your PRs again and creating an issue pointing out unhooked listeners like we did in Pyface, let's create an issue in traitsui now - which we can continually update?

Reply:
I've opened #1526